iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/hint/DocumentOrganizationBuilderTest.java

```diff
@@ -0,0 +1,184 @@
+
+    @Mock
+    private JavaRDD<ProjectOrganization> projectOrganizationRdd;
+
+    @Captor
+    private ArgumentCaptor<PairFunction<ProjectOrganization, String, ProjectOrganization>> pairProjectOrganizationFunctionArg;
+
+    @Mock
+    private JavaPairRDD<String, ProjectOrganization> projectOrganizationPairedRdd;
+
+    @Mock
+    private JavaPairRDD<String, Tuple2<DocumentProject, ProjectOrganization>> documentOrganizationJoinedRdd;
+
+    @Captor
+    private ArgumentCaptor<Function<Tuple2<String, Tuple2<DocumentProject, ProjectOrganization>>, DocumentOrganization>> mapDocumentOrganizationFunctionArg;
+
+    @Mock
+    private JavaRDD<DocumentOrganization> documentOrganizationMappedRdd;
+
+    @Mock
+    private JavaRDD<DocumentOrganization> documentOrganizationDistinctRdd;
+
+    private final String documentId = "docId";
+
+    private final String projectId = "projId";
+
+    private final String organizationId = "orgId";
+
+    private final Float confidenceLevel = 0.9f;
+
+    // ------------------------ TESTS --------------------------
+    
+    @Test(expected = NullPointerException.class)
+    public void build_NULL_DOCUMENT_PROJECT() {
+        // execute
+        builder.build(null, projectOrganizationRdd, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void build_NULL_PROJECT_ORGANIZATION() {
+        // execute
+        builder.build(documentProjectRdd, null, null);
+    }
+
+    @Test()
+    public void build_WITH_NULL_THRESHOLD() throws Exception {
+        // given
+        stub();
+        Float docProjConfidenceLevelThreshold = null;
+        // execute
+        JavaRDD<DocumentOrganization> result = builder.build(documentProjectRdd, projectOrganizationRdd,
+                docProjConfidenceLevelThreshold);
+        // assert
+        assertResults(result, docProjConfidenceLevelThreshold);
+    }
+
+    @Test()
+    public void build() throws Exception {
+        // given
+        stub();
+        Float docProjConfidenceLevelThreshold = 0.5f;
+        // execute
+        JavaRDD<DocumentOrganization> result = builder.build(documentProjectRdd, projectOrganizationRdd,
+                docProjConfidenceLevelThreshold);
+        // assert        
+        assertResults(result, docProjConfidenceLevelThreshold);
+    }
+
+    // ------------------------ PRIVATE --------------------------
+
+    private void stub() {
+        doReturn(documentProjectPairedRdd).when(documentProjectRdd).mapToPair(any());
+        doReturn(documentProjectPairedFilteredRdd).when(documentProjectPairedRdd).filter(any());
+        doReturn(projectOrganizationPairedRdd).when(projectOrganizationRdd).mapToPair(any());
+        doReturn(documentOrganizationJoinedRdd).when(documentProjectPairedFilteredRdd).join(any());
+        doReturn(documentOrganizationMappedRdd).when(documentOrganizationJoinedRdd).map(any());
+        doReturn(documentOrganizationDistinctRdd).when(documentOrganizationMappedRdd).distinct();
```

Comment:
Prefer Mockito.when instead of Mockito.doReturn. See the doc of Mockito.doReturn.